Add choose_qparams_per_token_asymmetric for llama on XNNPACK

andrewor14 opened the pull request:
Summary:
XNNPACK uses asymmetric activation quantizations,
but the existing `choose_qparams_per_token` assumed symmetric
quantization (zero point is always 0). This caused significant
numerical discrepancies between eager and lowered models.
This commit adds a new asymmetric version of `choose_qparams_per_token`
for this purpose.

Reviewed By: digantdesai

Differential Revision: D54323650

digantdesai reviewed (commented):
Watch out for CI failures though

andrewor14 committed:
Add choose_qparams_per_token_asymmetric for llama on XNNPACK

Summary:
XNNPACK uses asymmetric activation quantizations,
but the existing `choose_qparams_per_token` assumed symmetric
quantization (zero point is always 0). This caused significant
numerical discrepancies between eager and lowered models.
This commit adds a new asymmetric version of `choose_qparams_per_token`
for this purpose.

Reviewed By: digantdesai

Differential Revision: D54323650

cccclai commented:
Close stale PR
